src/app/problems/containers/problems/problems-page.component.spec.ts

```diff
@@ -100,4 +91,39 @@ describe('ProblemsPageComponent', () => {
             ])
         });
     });
+
+    describe('sortBySeverity', () => {
+        it('should sort by severity - Critical first, Warning last', () => {
+            const criticalProblem = problemGenerator(moment(), 'Critical')
+            const urgentProblem = problemGenerator(moment(), 'Urgent')
+            const warningProblem = problemGenerator(moment(), 'Warning')
+            const undefinedSeverityProblem = problemGenerator(moment(), 'SomeString')
+
+            const problems: Problem[] = [
+                urgentProblem,
+                warningProblem,
+                undefinedSeverityProblem,
+                criticalProblem
+            ]
+
+            const sortedProblems = problemsPageComponent.sortBySeverity(problems)
+            expect(sortedProblems).toEqual([
+                criticalProblem,
+                urgentProblem,
+                warningProblem,
+                undefinedSeverityProblem
+            ])
+        });
+    });
 });
+
+function problemGenerator(createdAt: moment.Moment, severity = 'Warning'): Problem {
```

Comment:
can we name it generateProblem, createProblem etc? function names should be verbs